lib/clouds/libcloud_common.py

```diff
@@ -820,143 +975,125 @@ def vmcreate(self, obj_attr_list) :
                 cbwarn(line, True)
             _status = 23
             _fmsg = str(e)
-            cbwarn("Error reaching cbtool: " + _fmsg)
-            for line in traceback.format_exc().splitlines() :
-                cberr(line, True)
+            cbwarn("Error reaching digitalocean: " + _fmsg)
 
         finally :
-            if "image" in obj_attr_list :
-                del obj_attr_list["image"]
-            
-            if _status :
-                _msg = "Instance " + obj_attr_list["uuid"] + " could not be created "
-                _msg += "on " + self.description + " \"" + obj_attr_list["cloud_name"] + "\" : "
-                _msg += _fmsg + " (The instance creation will be rolled back)"
-                cberr(_msg)
-
-                if "cloud_vm_uuid" in obj_attr_list :
-                    obj_attr_list["mgt_901_deprovisioning_request_originated"] = int(time())
-                    self.vmdestroy(obj_attr_list)
-                else :
-                    if _reservation :
-                        _reservation.destroy()
-
-                raise CldOpsException(_msg, _status)
-            else :
-                _msg = "Instance " + obj_attr_list["uuid"] + " was successfully "
-                _msg += "created on " + self.description + " \"" + obj_attr_list["cloud_name"]
-                _msg += "\"."
-                cbdebug(_msg)
-                return _status, _msg
+            if not _status :
+                if "instance_obj" in obj_attr_list :
+                    del obj_attr_list["instance_obj"]
+    
+                if obj_attr_list["cloud_vv_instance"] :
+                    del obj_attr_list["cloud_vv_instance"]
+               
+            _status, _msg = self.common_messages("VM", obj_attr_list, "created", _status, _fmsg)
+            return _status, _msg
 
     @trace
     def vmdestroy(self, obj_attr_list) :
-        _status = 100
-        _fmsg = "An error has occurred, but no error message was captured"
-        _wait = int(obj_attr_list["update_frequency"])
-        cbdebug("Last known state: " + str(obj_attr_list["last_known_state"]))
-        _curr_tries = 0
-        _max_tries = int(obj_attr_list["update_attempts"])
-        credentials_list = obj_attr_list["credentials_list"]
-
+        '''
+        TBD
+        '''
         try :
-            self.connect(credentials_list, obj_attr_list)
+            _status = 100
+            _fmsg = "An error has occurred, but no error message was captured"
 
-            _msg = "Sending a termination request for "  + obj_attr_list["name"]
-            _msg += " (cloud-assigned uuid " + obj_attr_list["cloud_vm_uuid"] + ")"
-            _msg += "...."
-            cbdebug(_msg, True)
+            cbdebug("Last known state: " + str(obj_attr_list["last_known_state"]))
 
-            firsttime = True
-            _time_mark_drs = int(time())
-            while True :
-                _errmsg = "get_vm_instance"
-                cbdebug("Getting instance...")
```

Comment:
This logic was very sensitive and difficult to get right in testing due to the way the libcloud adapter works. Was this logic preserved somewhere?